lib/Mailer/Methods/ErrorMappers/MailPoetMapper.php

```diff
@@ -50,19 +51,36 @@ function getErrorForResult(array $result, $subscribers) {
         $message = __('Email service is temporarily not available, please try again in a few minutes.', 'mailpoet');
         $retry_interval = self::TEMPORARY_UNAVAILABLE_RETRY_INTERVAL;
         break;
-      case API::RESPONSE_CODE_BANNED_ACCOUNT:
-        $message = Helpers::replaceLinkTags(
-          __('You currently are not permitted to send any emails with MailPoet Sending Service, which may have happened due to poor deliverability. Please [link]contact our support team[/link] to resolve the issue.', 'mailpoet'),
-          'https://www.mailpoet.com/support/',
-          array('target' => '_blank')
-        );
+      case API::RESPONSE_CODE_CAN_NOT_SEND:
+        if ($result['message'] === MailerError::MESSAGE_EMAIL_NOT_AUTHORIZED) {
```

Comment:
I think this whole code inside this `case` branch should be a separate method. This is very bad for understanding and readability.